libcontainer/standard_init_linux.go

```diff
@@ -189,14 +190,22 @@ func (l *linuxStandardInit) Init() error {
 			return err
 		}
 	}
-	if err := finalizeNamespace(l.config); err != nil {
+	if err := finalizeNamespace(l.config, l.addHome); err != nil {
 		return err
 	}
 	// finalizeNamespace can change user/group which clears the parent death
 	// signal, so we restore it here.
 	if err := pdeath.Restore(); err != nil {
 		return fmt.Errorf("can't restore pdeath signal: %w", err)
 	}
+
+	// In case we have any StartContainer hooks to run, and they don't
+	// have environment configured explicitly, make sure they will be run
+	// with the same environment as container's init.
+	if h := l.config.Config.Hooks[configs.StartContainer]; len(h) > 0 {
+		h.SetDefaultEnv(l.config.Env)
+	}
+
```

Comment:
@lifubang FYI this is the fix to the startContainer hook issue wrt environment you've reported back in June.

Reply:
Why added in here, but not together with the line before we prepare to run `StartContainer` hooks?
I mean put them before this line:
https://github.com/opencontainers/runc/blob/5a88394b1f68f1d5417b37485c9c26af91d1ccb6/libcontainer/standard_init_linux.go#L282